src/DataValues/BoundedQuantityValue.php

```diff
@@ -0,0 +1,381 @@
+	}
+
+	/**
+	 * Returns this quantity's lower bound.
+	 *
+	 * @since 0.1
+	 *
+	 * @return DecimalValue
+	 */
+	public function getLowerBound() {
+		return $this->lowerBound;
+	}
+
+	/**
+	 * Returns the size of the uncertainty interval.
+	 * This can roughly be interpreted as "amount +/- uncertainty/2".
+	 *
+	 * The exact interpretation of the uncertainty interval is left to the concrete application or
+	 * data point. For example, the uncertainty interval may be defined to be that part of a
+	 * normal distribution that is required to cover the 95th percentile.
+	 *
+	 * @since 0.1
+	 *
+	 * @return float
+	 */
+	public function getUncertainty() {
+		return $this->upperBound->getValueFloat() - $this->lowerBound->getValueFloat();
+	}
+
+	/**
+	 * Returns a DecimalValue representing the symmetrical offset to be applied
+	 * to the raw amount for a rough representation of the uncertainty interval,
+	 * as in "amount +/- offset".
+	 *
+	 * The offset is calculated as max( amount - lowerBound, upperBound - amount ).
+	 *
+	 * @since 0.1
+	 *
+	 * @return DecimalValue
+	 */
+	public function getUncertaintyMargin() {
+		$math = new DecimalMath();
+
+		$lowerMargin = $math->sum( $this->amount, $this->lowerBound->computeComplement() );
+		$upperMargin = $math->sum( $this->upperBound, $this->amount->computeComplement() );
+
+		$margin = $math->max( $lowerMargin, $upperMargin );
+		return $margin;
+	}
+
+	/**
+	 * Returns the order of magnitude of the uncertainty as the exponent of
+	 * last significant digit in the amount-string. The value returned by this
+	 * is suitable for use with @see DecimalMath::roundToExponent().
+	 *
+	 * @example: if two digits after the decimal point are significant, this
+	 * returns -2.
+	 *
+	 * @example: if the last two digits before the decimal point are insignificant,
+	 * this returns 2.
+	 *
+	 * Note that this calculation assumes a symmetric uncertainty interval,
+	 * and can be misleading.
+	 *
+	 * @since 0.1
+	 *
+	 * @return int
+	 */
+	public function getOrderOfUncertainty() {
+		// the desired precision is given by the distance between the amount and
+		// whatever is closer, the upper or lower bound.
+		//TODO: use DecimalMath to avoid floating point errors!
+		$amount = $this->amount->getValueFloat();
+		$upperBound = $this->upperBound->getValueFloat();
+		$lowerBound = $this->lowerBound->getValueFloat();
+		$precision = min( $amount - $lowerBound, $upperBound - $amount );
+
+		if ( $precision === 0.0 ) {
+			// If there is no uncertainty, the order of uncertainty is a bit more than what we have digits for.
+			return -strlen( $this->amount->getFractionalPart() );
+		}
+
+		// e.g. +/- 200 -> 2; +/- 0.02 -> -2
+		// note: we really want floor( log10( $precision ) ), but have to account for
+		// small errors made in the floating point operations above.
+		// @todo: use bcmath (via DecimalMath) to avoid this if possible
+		$orderOfUncertainty = floor( log10( $precision + 0.0000000005 ) );
```

Comment:
Let's please not rewrite this odd code, it was rather tricky to get right; we can factor it out into a separate class, but let's do that in a separate change.

Let's also keep in mind that the transform() method is central to unit conversion. Even if it's not really used at the moment, we should keep it around.